Purchases/Public/RCPurchases.m

```diff
@@ -644,21 +645,34 @@ - (void)paymentDiscountForProductDiscount:(SKProductDiscount *)discount
     }];
 }
 
+- (void)invalidatePurchaserInfoCache {
+    RCDebugLog(@"Purchaser info cache is invalidated");
+    [self.deviceCache clearPurchaserInfoCacheTimestamp];
+}
+
 #pragma mark - Private Methods
 
 - (void)applicationDidBecomeActive:(__unused NSNotification *)notif
 {
     RCDebugLog(@"applicationDidBecomeActive");
-    if ([self.deviceCache isCacheStale]) {
-        RCDebugLog(@"Cache is stale, updating caches");
-        [self updateCachesWithCompletionBlock:^(RCPurchaserInfo *info, NSError *error) {
-            if (info) {
-                [self sendUpdatedPurchaserInfoToDelegateIfChanged:info];
-            }
-        }];
+    if ([self.deviceCache isPurchaserInfoCacheStale]) {
+        RCDebugLog(@"PurchaserInfo cache is stale, updating caches");
+        [self updateCacheAndSendUpdatedPurchaserInfoIfChanged];
+    }
+    if ([self.deviceCache isOfferingsCacheStale]) {
+        RCDebugLog(@"Offerings cache is stale, updating caches");
+        [self updateOfferingsCache:nil];
     }
 }
 
+- (void)updateCacheAndSendUpdatedPurchaserInfoIfChanged {
+    [self fetchAndCachePurchaserInfoWithCompletion:^(RCPurchaserInfo *info, NSError *error) {
+        if (info) {
```

Comment:
I was comparing this to the Android implementation and realized that `fetchAndCachePurchaserInfoWithCompletion` is actually doing `sendUpdatedPurchaserInfoToDelegateIfChanged:info` inside. I believe this method can be removed and directly replaced with `fetchAndCachePurchaserInfoWithCompletion`.